### ISSUE TITLE
Allow vanilla tools to keep enchants upon being used in the crafting recipe.

### ISSUE THREAD

**Issue by ItzASecret** [PluginBugs/Issues-ItemsAdder#1759]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same suggestion was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if the feature is already implemented.
- [X] I already searched on the **Discord** server to know if anyone already has a solution for this or if the feature was already suggested.

### Discord tag (optional)

ItzASecretBoi#1300

### Describe the solution you'd like

Some way of making items keep enchants when being used in custom crafting recipes in the config. Perhaps some sort of whitelist for all tools in the config would work? Or a true/false option for each tool in the config. Whatever works.

It would also be nice if this was possible using a smithing table, however I see that a GitHub request asking the same thing has already been labeled "Future Update" on Nov 21, 2020.

The other github request I mentioned - - - > https://github.com/PluginBugs/Issues-ItemsAdder/issues/524

### Is your feature request related to a problem?

No.

### Describe alternatives you've considered

I considered just giving players the enchantment books back via disenchantment with another plugin, but I thought I'd open a feature request instead.

### Additional context

Thanks for reading!

**Comment by ItzASecret** [comment]
Oh and i forgot, the idea should apply to armors too, pretty much anything enchantable. shears, etc.

**Comment by LoneDev6** [comment]
I think that adding this kind of functionality to crafting table is just confusing.
Currently you should already be able [to do that](https://github.com/PluginBugs/Issues-ItemsAdder/issues/524#issuecomment-877481770), the only issue is that it will work only on unmodified items (items with no damage, not renamed in anvil etc).
This is a limitation of the trade GUI of the game itself and I have absolutely NO WAY to fix that.


I'd prefer to implement the Smithing Table recipes, the only issue is that the Smithing Table has no recipebook attached, so it will be a bit difficult for players to learn recipes, I think.

I wait for some other suggestions for that task, thanks for the ideas!

**Comment by ItzASecret** [comment]
Well I dont know if it would help but I think I had an idea that could solve this a little, I opened a feature request where I suggested custom informational items we could display on the /ia menu that could help players convey messages through the menu. Stuff like where to/how to craft things as you said, unless I read it wrong lol. Its the same thing as the creeper head you use to show what mobs drop what items. 

Here it is if your interested:

https://github.com/PluginBugs/Issues-ItemsAdder/issues/1764
